common/libs/network_protocol/network_protocol.proto

```diff
@@ -1,128 +1,202 @@
-  GET_KD = 0x25;       // Set Kd (PID differentiation coefficient
-  SET_BLOWER = 0x26;   // Set blower PWM signal value 0x00 to 0x0FF; 0x00 - off,
-                       // 0xFF - full speed
-  RESET = 0x27;        // Reset Ventilator Controller board
-  SET_SOLENOID = 0x28; // Set solenoid normal state 0x00 - Normally open, 0x01
-                       // Normally closed
-
-  // Mixed ENGINEERING/MEDICAL mode commands
-
-  SET_TX_RATE = 0x40; // Set periodic data transmission rate - to be defined
-  GET_TX_RATE = 0x41; // Get periodic data transmission rate - to be defined
-  SET_CONTROLLER_MODE = 0x42; // Set Ventilator Controller mode; 0x00 - Medical
-                              // Mode, 0xFF - Engineering Mode
-
-  GET_CONTROLLER_MODE = 0x43; // Get Ventilator Controller mode
-  COMM_CHECK = 0x44;          // Send communications check
-  SET_VENTILATOR_MODE = 0x45; // Set ventilator mode; 0x00 - PRVC, 0xFF - ACV
-  GET_VENTILATOR_MODE = 0x46; // Get ventilator mode; 0x00 - PRVC, 0xFF - ACV
-  START_VENTILATOR = 0x47;    // Ventilator start
-  STOP_VENTILATOR = 0x48;     // Ventilator stop
-  NONE = 0xFF; // For Controller originating packets that are not responses to
-               // commands
-}
+// Periodically sent from the controller to the GUI.
+message ControllerStatus {
+  // milliseconds since controller started up.
+  required uint64 uptime_ms = 1;
 
-// Sent form GUI to controller
-message Command {
-  required CommandType cmd = 1;
-  optional float data = 2;
-}
+  // Current params being used by the the controller.  This is used to ACK
+  // params sent by the GUI.
+  required VentParams params = 2;
 
-// Sent from GUI in response to Alarm ControllerData packet
-message GuiAck { required uint32 type = 1; }
+  // Current sensor readings.
+  required SensorReadings sensor_readings = 3;
 
-// Periodic controller status sent in ControllerData packet
-message ControllerStatus {
-  required uint64 time = 1;
-  required float pressure = 2;
-  required float volume = 3;
-  required float flow = 4;
-  optional uint32 alarm_flags = 5;
-}
+  // Active alarms fired by the controller.
+  //
+  // TODO: The max number of alarms wasn't chosen carefully.
+  // The max here should match GuiStatus.controller_alarms's max.
+  repeated Alarm controller_alarms = 4 [ (nanopb).max_count = 4 ];
 
-// Sent in ControllerData packet on controller reset
-message ControllerIdentification {
-  required uint64 time = 1;
-  required uint32 version_fw = 2;
-  required uint32 version_hw = 3;
-  optional string about = 4;
+  // TODO: Include some sort of code version, e.g. git sha that the controller
+  // was built from?
 }
 
-enum AlarmType { TO_BE_DEFINED = 0; }
+// Values set by the ventilator operator.
+message VentParams {
+  required VentMode mode = 1;
 
-enum AlarmLevel {
-  LOW = 0;
-  MEDIUM = 1;
-  HIGH = 2;
-}
+  required uint32 percent_inspiratory_o2 = 2; // FiO2 - frac inspiratory oxygen
```

Comment:
Are there any invariants like "if a parameter is irrelevant to this mode, then it is zero"? Would it make sense to use oneofs per mode?